Add Nix files for reproducible builds

john-shaffer opened the pull request:
This adds a few files that add Nix dependencies pinned to a git sha of the Nix repo. The idea is that Nix gives you the exact version of bash, php, composer, zip, etc. specified so that builds are reproducible and those dependencies are version controlled (by the git sha in sources.json).

The files have no effect for someone not using Nix and can be safely ignored.

leonstafford commented:
Supporting Nix seems worth compromise of adding more things to the repo.

Could we please get some usage instructions to go along with this, though? ie, link to the GH/wp2static repo with the nix integration test stuff in it or in the wp2static-www as a guide for developers?

leonstafford commented:
And we'll want to add similar to all add-on repos?

leonstafford commented on nix/sources.json:
```diff
@@ -0,0 +1,14 @@
+        "sha256": "1xlgynfw9svy7nvh9nkxsxdzncv9hg99gbvbwv3gmrhmzc3sar75",
+        "type": "tarball",
+        "url": "https://github.com/NixOS/nixpkgs/archive/5f244caea76105b63d826911b2a1563d33ff1cdc.tar.gz",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
```
~~placeholder?~~ ugh

john-shaffer committed:
Add Nix files for reproducible builds

john-shaffer committed:
Use the more portable /usr/bin/env bash instead of /bin/bash

/usr/bin/env is part of the POSIX standard and is present on
OSes like BSDs and NixOS where /bin/bash isn't.

john-shaffer commented:
I simplified the Nix files down to one small file and added shellcheck.

I will try to get some docs in when I get a chance.
